File: backend/micro_epoweredmove_vehicle/src/main/java/com/kosdiam/epoweredmove/models/enums/VehicleType.java
package com.kosdiam.epoweredmove.models.enums;

public enum VehicleType {
    CAR("Car"),
    MOTORBIKE("Motorbike"),
    SCOOTER("Scooter"),
    BICYCLE("Bicycle"),
    OTHER("Other");

    private String vehicleType;
    private VehicleType(String vehicleType){
        this.vehicleType = vehicleType;
    }

    @Override
    public String toString(){
        return vehicleType;
    }
}

File: frontend/E-PoweredMoveAndroidApp/app/src/main/java/com/kosdiam/epoweredmove/models/enums/VehicleType.kt
package com.kosdiam.epoweredmove.models.enums

import com.kosdiam.epoweredmove.R
import com.kosdiam.epoweredmove.models.interfaces.IEnumName

enum class VehicleType (override val stringResource: Int) : IEnumName {
    CAR(R.string.vehicle_type_car),
    MOTORBIKE(R.string.vehicle_type_motorbike),
    SCOOTER(R.string.vehicle_type_scooter),
    BICYCLE(R.string.vehicle_type_bicycle),
    OTHER(R.string.vehicle_type_other);
}